go/cmd/vttablet/vttablet.go

```diff
@@ -67,13 +67,29 @@ func main() {
 	if err != nil {
 		log.Exitf("failed to parse -tablet-path: %v", err)
 	}
+	if tabletAlias.Uid < 0 {
+		log.Exitf("invalid tablet id: %d", tabletAlias.Uid)
+	}
 
-	mycnf, err := mysqlctl.NewMycnfFromFlags(tabletAlias.Uid)
-	if err != nil {
-		log.Exitf("mycnf read failed: %v", err)
+	var mycnf *mysqlctl.Mycnf
+	var socketFile string
+	// If no connection parameters were specified, load the mycnf file
+	// and use the socket from it. If connection parameters were specified,
+	// we assume that the mysql is not local, and we skip loading mycnf.
+	// This also means that backup and restore will not be allowed.
+	if !dbconfigs.HasConnectionParams() {
+		var err error
+		if mycnf, err = mysqlctl.NewMycnfFromFlags(tabletAlias.Uid); err != nil {
+			log.Exitf("mycnf read failed: %v", err)
+		}
+		socketFile = mycnf.SocketFile
+	} else {
+		log.Info("connection parameters were specified. Not loading my.cnf.")
 	}
 
-	dbcfgs, err := dbconfigs.Init(mycnf.SocketFile)
+	// If connection parameters were not specified, we'll use
+	// mycnf's socket file (if present).
+	dbcfgs, err := dbconfigs.Init(socketFile)
```

Comment:
Can you change the comment to make it explicit that socketFile is unitialiazed here in the case dbconfigs.HasConnectionParams, but this is fine because it will be used only in the case where dbconfigs.HasConnectionParams is False and that case has been dealt with above.